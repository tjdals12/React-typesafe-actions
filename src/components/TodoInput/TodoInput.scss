@import 'utils';

.todo-input {
    display: flex;
    border-top: 1px solid $oc-gray-3;
    border-bottom: 2px solid $oc-gray-3;
    padding: 1rem;

    .input {
        flex: 1;
        border: none;
        border-bottom: 1px solid $oc-gray-3;
        outline: none;
        font-size: 1.25rem;

        &:focus {
            transition: all 0.25s ease-in;
            border-bottom: 1px solid $oc-cyan-6;
        }

        &::placeholder {
            color: $oc-gray-5;
            font-style: italic;
        }
    }

    .button {
        width: 5rem;
        height: 3rem;

        margin-left: 1rem;

        background: $oc-cyan-6;
        color: white;
        font-size: 1.25rem;
        border: none;
        outline: none;
        border-radius: 0.5rem;

        cursor: pointer;

        &:hover {
            transition: all 0.25s ease-in;
            background: $oc-cyan-4;
        }

        &:active {
            background: $oc-cyan-9;
        }
    }
}
