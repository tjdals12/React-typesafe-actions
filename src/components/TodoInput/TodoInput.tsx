import React, { useState, FormEvent } from 'react';
import classNames from 'classnames/bind';
import useAddTodo from 'hooks/useAddTodo';
import styles from './TodoInput.scss';

const cx = classNames.bind(styles);

export default function TodoInput(): JSX.Element {
    const [value, setValue] = useState('');
    const addTodo = useAddTodo();

    const onSubmit = (e: FormEvent): void => {
        e.preventDefault();
        addTodo(value);
        setValue('');
    };

    return (
        <form className={cx('todo-input')} onSubmit={onSubmit}>
            <input
                className="input"
                placeholder="할 일을 입력하세요."
                value={value}
                onChange={(e): void => setValue(e.target.value)}
            />
            <button className="button" type="submit">
                등록
            </button>
        </form>
    );
}
