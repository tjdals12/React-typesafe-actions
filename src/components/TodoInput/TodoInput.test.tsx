import React from 'react';
import { render } from '@testing-library/react';
import TodoInput from './TodoInput';

describe('<TodoInput />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoInput />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has a input and a button', () => {
        const utils = render(<TodoInput />);
        expect(utils.container.querySelector('.input')).not.toBeNull();
        expect(utils.container.querySelector('.button')).not.toBeNull();
    });
});
