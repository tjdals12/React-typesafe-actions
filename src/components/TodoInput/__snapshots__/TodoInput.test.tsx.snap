// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoInput /> matches snapshot 1`] = `
<div>
  <form
    class="todo-input"
  >
    <input
      class="input"
      placeholder="할 일을 입력하세요."
      value=""
    />
    <button
      class="button"
      type="submit"
    >
      등록
    </button>
  </form>
</div>
`;
